### ISSUE TITLE
remote build: stop raising internal errors

### ISSUE THREAD

**Issue by mr-cal** [canonical/snapcraft#4908]
### What needs to get done

Remote builder errors should be caught and include a resolution and a link to the documentation

### Why it needs to get done

Remote build errors are raised as `snapcraft internal error`s and are do not help the user fix the problem.

source: https://matrix.to/#/!GGqzbFAUQprdPgYYCM:ubuntu.com/$eFv8F6yWzVPWZHp4eXV7gKHTh125YamJUJEnrkI2eC0?via=ubuntu.com&via=matrix.org&via=kde.org

**Comment by mr-cal** [comment]
Example:
https://github.com/snapcrafters/webkitgtk-sdk/actions/runs/9877563264/job/27366291368
```
Initialized empty Git repository in /home/ubuntu/actions-runner/_work/webkitgtk-sdk/webkitgtk-sdk/webkitgtk-6-gnome-2404-sdk/.git/
remote-build is experimental and is subject to change. Use with caution.
Starting new build. It may take a while to upload large projects.
Enumerating objects: 6, done.
Counting objects:  16% (1/6)
Counting objects:  33% (2/6)
Counting objects:  50% (3/6)
Counting objects:  66% (4/6)
Counting objects:  83% (5/6)
Counting objects: 100% (6/6)
Counting objects: 100% (6/6), done.
Delta compression using up to 4 threads
Compressing objects:  16% (1/6)
Compressing objects:  33% (2/6)
Compressing objects:  50% (3/6)
Compressing objects:  66% (4/6)
Compressing objects:  83% (5/6)
Compressing objects: 100% (6/6)
Compressing objects: 100% (6/6), done.
Writing objects:  16% (1/6)
Writing objects:  33% (2/6)
Writing objects:  50% (3/6)
Writing objects:  66% (4/6)
Writing objects:  83% (5/6)
Writing objects: 100% (6/6)
Writing objects: 100% (6/6), 3.13 KiB | 3.13 MiB/s, done.
Total 6 (delta 0), reused 0 (delta 0), pack-reused 0
remote:
remote: Create a merge proposal for 'main' on Launchpad by visiting:
remote:       https://code.launchpad.net/~snapcrafters-bot/snapcrafters-bot-craft-remote-build/+git/snapcraft-webkitgtk-6-gnome-2404-sdk-62339028b847a9b6959f212cb0dbab94/+ref/main/+register-merge
remote:
To https://git.launchpad.net/~snapcrafters-bot/snapcrafters-bot-craft-remote-build/+git/snapcraft-webkitgtk-6-gnome-2404-sdk-62339028b847a9b6959f212cb0dbab94
 + 4af9968...8fb9f64 master -> main (forced update)
snapcraft internal error: BadRequest()
Full execution log: '/home/ubuntu/.local/state/snapcraft/log/snapcraft-20240712-094551.019918.log'
cat: 'webkitgtk-6-gnome-2404-sdk_amd64*.txt': No such file or directory
Could not find build log
Could not find webkitgtk-6-gnome-2404-sdk_2.44.2_amd64.snap
```

**Comment by mr-cal** [comment]
Fixed via #5052
